src/main/kotlin/account/Account.kt

```diff
@@ -5,44 +5,35 @@ class Account(
     val conta: Int,
     val saldoInicial: Double
 ) {
-    var saldo = 0.0
+    var saldo  = saldoInicial
         protected set
     //protected faz com o valor do saldo nao seja modificado em outros arquivos, mas permite que as classes "filhas"
     // a utilizem
 
     fun deposita(valor: Double){
 
-        if(valor <= 0.0){
-            this.saldo += valor + saldoInicial
-        }
-        else if(valor > 0 && valor <= 10_000){
+        if(valor > 0.0 && valor <= 10_000.0){
             this.saldo += valor
         } else{
             throw error("valor não permitido")
         }
     }
 
     fun saque(valor: Double){
-        if(saldo >= valor){
-            this.saldo -= valor;
-            if(saldo < 0){
-                throw error("valor para saque não permitido")
-                this.saldo += valor;
+        if(valor <= saldo && valor > 0.0){
+                this.saldo -= valor
+            }
+            else{
+                throw error("valor não permitido")
             }
-        } else{
-            throw error("valor para saque não permitido")
         }
-    }
 
-    fun transferencia(valor: Double, conta: Account){
+    fun transferencia(valor: Double, conta: Account) {
         if (saldo > 0){
-            this.saldo -= valor
-            if(saldo < 0){
-                throw error("valor para tranferência não permitida")
-                this.saldo += valor;}
-            conta.deposita(valor)
-        } else {
+            saque(valor)
+            conta.deposita(valor)}
+        else {
             throw error("saldo em conta insuficiente")
         }
     }
-}
\ No newline at end of file
+}
```

Comment:
Questão bônus, depois que terminar os ajustes 🔥 

Se tivermos 3 métodos pra representar um caixa eletrônico:

```kotlin
fun abreGaveta() {
  // Abre gaveta do caixa eletrônico para receber ou devolver dinheiro
}
```

```kotlin
fun recebeDinheiro(valor: Double) {
  print("Sugando $valor reais do cliente...")
  if(valorErrado()) throw error("Valor recebido não confere")
  print("Valor conferido")
}
```

e

```kotlin
fun devolveDinheiro(valor: Double) {
  print("Devolvendo $valor reais ao cliente...")
}
```

Como ficaria a lógica nova de depósito e saque?

Reply:
@andreneery se for fazer esse desafio recomendo fazer em outro PR rsrs
Fica mais facil pra você e para a gente que vai fazer review :upside_down_face: